### ISSUE TITLE
[Bug] URL bar does not match website I am on

### ISSUE THREAD

**Issue by mattschlosser** [mozilla-mobile/fenix#17145]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
- Unsure. Needs investigation.

### Expected behavior
URL bar matches website I am on at all times

### Actual behavior
URL bar does not match website I am on in some cases

### Other info to help reproduce
- i was on search.twitter.com page last night. powered down device overnight. next morning, opened Facebook (can't remember if I clicked on the top site tile, or typed in the url bar, but given that the latest version loads the last tab I was on, I probably opened the app 
- tabs are set to automatically close after one day
- the website I am actually on (Facebook) correctly appears in history
- links won't open in new tabs (if i click on a link that wants to open in a new tab, it doesn't open, but it is added to the history)
- I can navigate to new pages within Facebook by clicking the links, new pages are correctly added to history, but URL bar does not change
- closing and reloading the app opened the search.twitter.com page again

### Screenshots
![Screenshot_20201219-071207](https://user-images.githubusercontent.com/8870594/102692816-2fc0c700-41d3-11eb-9cb2-42852c85f7f1.png)
![Screenshot_20201219-072420](https://user-images.githubusercontent.com/8870594/102692819-30f1f400-41d3-11eb-8c2d-e600949f4475.png)
![Screenshot_20201219-074826](https://user-images.githubusercontent.com/8870594/102692820-32232100-41d3-11eb-8ee5-be14e7cf8630.png)
![Screenshot_20201219-075608](https://user-images.githubusercontent.com/8870594/102692821-32232100-41d3-11eb-8742-46e74dfeaae4.png)

### Device information
Android 11 (RQ1A.201205.008)
* Android device: Pixel 4 XL
* Fenix version: 84.1.1

**Comment by mattschlosser** [comment]
Further details: I can reproduce this bug *sometimes* by opening Firefox just after unlocking my phone after my device has just started. Iinstead of showing the last tab that was loaded, it shows the new tab page, and clicking on any top site loads that top site, but the url bar still shows whatever last tab was loaded.

**Comment by sflorean** [comment]
@mattschlosser this is still reproducible for you?
I tested on Nightly 7/21 and couldn't reproduce it. Devices: Pixel 4XL (Android 11) and LG G7 Fit (Android 8.1).

**Comment by mattschlosser** [comment]
Happened again today...
![Screenshot_20220214-085909](https://user-images.githubusercontent.com/8870594/153903718-8ef5d512-c564-4e93-8542-a385d0ec3a57.png)
Firefox 97.1.0
Pixel 4 XL - Feb 2022 Android 12

This happened directly after I closed another tab. Then I switched to worldwaterfall tab. It appeared to try to load the page but failed. Switching to one of the other tabs I had open (e.g. Wordle) and then back would just keep showing the previous tab. I switched to a third tab (a search on duckduckgo) and then back to the worldwaterfall tab and it showed the duckduckgo search with the worldwaterfall url beneath.  I quit firefox and restarted it and the worldwaterfall tab then loaded normally.
